openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -524,10 +535,35 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
                 texture_hash
             ]
             if linearize:
-                self.log.info("tx: converting sRGB -> linear")
-                additional_args.extend(["--colorconvert", "sRGB", "linear"])
-
-            config_path = get_ocio_config_path("nuke-default")
+                if cmds.colorManagementPrefs(query=True, cmEnabled=True):
+                    render_colorspace = cmds.colorManagementPrefs(query=True, renderingSpaceName=True)  # noqa
+                    color_space_attr = resource["node"] + ".colorSpace"
+                    try:
+                        color_space = cmds.getAttr(color_space_attr)
+                    except ValueError:
+                        # node doesn't have color space attribute
+                        img_info = image_info(filepath)
+                        color_space = guess_colorspace(img_info)
+                    self.log.info("tx: converting {0} -> {1}".format(color_space, render_colorspace))  # noqa
+                    additional_args.extend(["--colorconvert",
+                                            color_space,
+                                            render_colorspace])
+                else:
+                    img_info = image_info(filepath)
+                    color_space = guess_colorspace(img_info)
+                    if color_space == "sRGB":
+                        self.log.info("tx: converting sRGB -> linear")
+                        additional_args.extend(["--colorconvert",
+                                                "sRGB",
+                                                "Raw"])
+                    else:
+                        self.log.info("tx: texture's colorspace "
+                                      "is already linear")
+
+            config_path = cmds.colorManagementPrefs(query=True,
+                                                    configFilePath=True)
+            if not os.path.exists(config_path):
+                raise RuntimeError("No OCIO config path found!")
```

Comment:
Seems odd we're doing this check after all the logic above, no?

Also, process texture is always called. Does this change now mean we always require a OCIO config?

Reply:
The ocio config file is needed for the "--colorconvert" argument if i remember correctly, maybe I move the related code into the if-condtion of the colorspace